alpaca/common/rest.py

```diff
@@ -185,20 +184,23 @@ def _data_get(
 
         path = f"/{endpoint_base}"
 
-        if isinstance(symbol_or_symbols, str) and symbol_or_symbols:
+        multi_symbol = not isinstance(symbol_or_symbols, str)
+
+        if not multi_symbol and symbol_or_symbols:
             path += f"/{symbol_or_symbols}"
         else:
             data["symbols"] = ",".join(symbol_or_symbols)
 
         if endpoint:
             path += f"/{endpoint}"
 
-        resp_grouped_by_symbol = not isinstance(symbol_or_symbols, str)
-
         while True:
 
             actual_limit = None
+
+            # adjusts the limit parameter value if it is over the page_limit
             if limit:
+                # actual_limit is the adjusted total number of items to query per request
```

Comment:
The tests are failing here where we don't specify limit. Maybe this should default to `page_limit` instead of None? :thinking:

Reply:
Ah I think my confusion is rising entirely out of that we have limit and page_limit both in here, and this might also lead to confusion for users comparing it to the raw http values (we also need to change this to ensure page_limit is not above DATA_V2_MAX_LIMIT).  It'll also cause confusion when we bring the returning an iterator changes to these if they wanted to say get `limit` sized pages each time out of the iterator which is how the api itself returns pages.

The limit here is for total number of items returned when asking for a flat list of results not the actual api's `limit` parameter even though that's what we end up using it for :thinking: .

Reply:
Yeah I was wondering why there was an explicit `page_limit` parameter. It seemed to me that it should just always default to DATA_V2_MAX_LIMIT`. I thought maybe it was included for some edge cases where you might want to lower that limit.

Maybe we can remove the `page_limit` parameter from the method for now until an obvious use case arises?

Reply:
ah see to me the obvious case is already here with the wanting an iterator for pages. :thinking: